File: models/class_item.py
from class_ModeloJSON import ModeloJSON, json


class Item:
    def __init__(self, id, id_Produto, id_KitSolar, quantidade):
        self.__id = 0
        self.__id_Produto = 0
        self.__id_KitSolar = 0
        self.__quantidade= 0
     
    
    # sets
        self.set_id(id)
        self.set_id_Produto(id_Produto)
        self.set_id_KitSolar(id_KitSolar)
        self.set_quantidade(quantidade)

    def set_id(self, id):
        if ((type(id)  == int) and id >= 0):
            self.__id = id
        else:
            raise ValueError("Valor inválido, tente outro valor positivo")
    
    def set_id_Produto(self, id_Produto):
        if ((type(id_Produto)  == int) and id_Produto >= 0):
            self.__id_Produto = id_Produto
        else:
            raise ValueError("Valor inválido, tente outro valor positivo")
    
    def set_id_KitSolar(self, id_KitSolar):
        if ((type(id_KitSolar)  == int) and id_KitSolar >= 0):
            self.__id_KitSolar = id_KitSolar
        else:
            raise ValueError("Valor inválido, tente outro valor positivo")

    def set_quantidade(self,quantidade):
        if (type(quantidade) == int and quantidade >= 0):
            self.__quantidade = quantidade
        else:
            raise ValueError("Valor inválido, tente outro valor positivo")



    # metodos gets
    def get_id(self):
        return self.__id

    def get_id_Produto(self):
        return self.__id_Produto

    def get_id_KitSolar(self):
        return self.__id_KitSolar

    def get_quantidade(self):
        return self.__quantidade
    
    def __str__(self):
        return f" Id: {self.get_id()} - ID-Produto: {self.get_id_Produto()} - ID-kitSolar: {self.get_id_KitSolar()} - Quantidade: {self.get_quantidade()} "

    # def to_dict(self):
       
    #     return {
    #         "id": self.get_id(),
    #         "id_Produto": self.get_id_Produto(),
    #         "id_KitSolar": self.get_id_KitSolar(),
    #         "quantidade": self.get_quantidade()
            
    #     }

    @classmethod
    def from_dict(cls, data):
        
        return cls(
            id=data["id"],
            id_Produto=data["id_Produto"],
            id_KitSolar=data["id_KitSolar"],
            quantidade=data["quantidade"]
        )

class Itens(ModeloJSON):
    arquivo_json = "admin/model/Itens.json"

    @classmethod
    def inserir(cls, item):
        super().inserir(item)

    @classmethod
    def listar(cls):
        return super().listar()

    @classmethod
    def listar_id(cls, id):
        return super().listar_id(id)

    @classmethod
    def atualizar(cls, item):
        super().atualizar(item)

    @classmethod
    def excluir(cls, item):
       super().excluir(item)

    @classmethod
    def abrir(cls):
        
        cls.lista_obj = []
        try:
            with open(cls.arquivo_json, "r") as arquivo:
                arquivo_json = json.load(arquivo)
                cls.lista_obj = [Item.from_dict(obj) for obj in arquivo_json]
        except FileNotFoundError:
            pass

    @classmethod
    def salvar(cls):
        with open(cls.arquivo_json, "w") as arquivo:
            json.dump([obj.to_dict() for obj in cls.lista_obj], arquivo, indent=4)


objeto = Item(0,1,2,30)

mostrar = Itens.listar()
for i in mostrar:
    print(i)

File: models/ClassItem.java
import org.json.JSONArray;
import org.json.JSONObject;

import java.io.File;
import java.io.FileWriter;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;

public class ClassItem {

    // Class representing an Item
    public static class Item {
        private int id;
        private int idProduto;
        private int idKitSolar;
        private int quantidade;

        // Constructor
        public Item(int id, int idProduto, int idKitSolar, int quantidade) {
            this.setId(id);
            this.setIdProduto(idProduto);
            this.setIdKitSolar(idKitSolar);
            this.setQuantidade(quantidade);
        }

        // Setters with validation
        public void setId(int id) {
            if (id >= 0) {
                this.id = id;
            } else {
                throw new IllegalArgumentException("ID must be a non-negative integer.");
            }
        }

        public void setIdProduto(int idProduto) {
            if (idProduto >= 0) {
                this.idProduto = idProduto;
            } else {
                throw new IllegalArgumentException("ID Produto must be a non-negative integer.");
            }
        }

        public void setIdKitSolar(int idKitSolar) {
            if (idKitSolar >= 0) {
                this.idKitSolar = idKitSolar;
            } else {
                throw new IllegalArgumentException("ID Kit Solar must be a non-negative integer.");
            }
        }

        public void setQuantidade(int quantidade) {
            if (quantidade >= 0) {
                this.quantidade = quantidade;
            } else {
                throw new IllegalArgumentException("Quantidade must be a non-negative integer.");
            }
        }

        // Getters
        public int getId() {
            return id;
        }

        public int getIdProduto() {
            return idProduto;
        }

        public int getIdKitSolar() {
            return idKitSolar;
        }

        public int getQuantidade() {
            return quantidade;
        }

        // Convert to a JSON-like Map
        public JSONObject toDict() {
            JSONObject obj = new JSONObject();
            obj.put("id", id);
            obj.put("idProduto", idProduto);
            obj.put("idKitSolar", idKitSolar);
            obj.put("quantidade", quantidade);
            return obj;
        }

        // Create an Item from a JSON-like Map
        public static Item fromDict(JSONObject data) {
            return new Item(
                    data.getInt("id"),
                    data.getInt("idProduto"),
                    data.getInt("idKitSolar"),
                    data.getInt("quantidade")
            );
        }

        @Override
        public String toString() {
            return "Item { " +
                    "ID=" + id +
                    ", ID Produto=" + idProduto +
                    ", ID Kit Solar=" + idKitSolar +
                    ", Quantidade=" + quantidade +
                    " }";
        }
    }

    // Class managing a list of Items and saving/loading to/from JSON
    public static class Itens {
        private static final String ARQUIVO_JSON = "itens.json";
        private static List<Item> listaItens = new ArrayList<>();

        // Insert an item
        public static void inserir(Item item) {
            listaItens.add(item);
        }

        // List all items
        public static List<Item> listar() {
            return listaItens;
        }

        // Find an item by ID
        public static Item listarPorId(int id) {
            for (Item item : listaItens) {
                if (item.getId() == id) {
                    return item;
                }
            }
            return null;
        }

        // Update an item
        public static void atualizar(Item itemAtualizado) {
            for (int i = 0; i < listaItens.size(); i++) {
                if (listaItens.get(i).getId() == itemAtualizado.getId()) {
                    listaItens.set(i, itemAtualizado);
                    break;
                }
            }
        }

        // Delete an item
        public static void excluir(int id) {
            listaItens.removeIf(item -> item.getId() == id);
        }

        // Load items from JSON file
        public static void abrir() {
            listaItens.clear();
            File file = new File(ARQUIVO_JSON);
            if (!file.exists()) {
                System.out.println("JSON file not found. Creating a new one.");
                return;
            }

            try {
                String content = new String(Files.readAllBytes(Paths.get(ARQUIVO_JSON)));
                JSONArray jsonArray = new JSONArray(content);
                for (int i = 0; i < jsonArray.length(); i++) {
                    JSONObject obj = jsonArray.getJSONObject(i);
                    Item item = Item.fromDict(obj);
                    listaItens.add(item);
                }
                System.out.println("Items loaded successfully!");
            } catch (Exception e) {
                System.err.println("Error loading items: " + e.getMessage());
            }
        }

        // Save items to JSON file
        public static void salvar() {
            JSONArray jsonArray = new JSONArray();
            for (Item item : listaItens) {
                jsonArray.put(item.toDict());
            }

            try (FileWriter file = new FileWriter(ARQUIVO_JSON)) {
                file.write(jsonArray.toString(4));
                System.out.println("Items saved successfully!");
            } catch (Exception e) {
                System.err.println("Error saving items: " + e.getMessage());
            }
        }
    }

    // Main method for testing
    public static void main(String[] args) {
        // Load items from JSON (if any)
        Itens.abrir();

        // Create a new item
        Item item1 = new Item(0, 1, 2, 30);
        Itens.inserir(item1);

        // Save items to JSON
        Itens.salvar();

        // List all items
        System.out.println("Items saved:");
        for (Item item : Itens.listar()) {
            System.out.println(item);
        }
    }
}